cli-toolchain/graphql.md

```diff
@@ -2379,6 +2379,160 @@ Here is a complete list of searchable operations per GraphQL type supported as o
+  speakTranslatedImageText: String @predictions(actions: [
+    identifyText
+    translateText
+    convertTextToSpeech
+  ])
+}
+```
+
+An example of that query will look like:
+
+```graphql
+query SpeakTranslatedImageText($input: SpeakTranslatedImageTextInput!) {
+  speakTranslatedImageText(input: {
+    identifyText: {
+      key: "myimage.jpg"
+    }
+    translateText: {
+      sourceLanguage: "en"
+      targetLanguage: "es"
+    }
+    convertTextToSpeech: {
+      voiceID: "Conchita"
+    }
+  })
+}
+```
+
+A code example of this using the JS Library:
+```js
+import React, { useState } from 'react';
+import API, { graphqlOperation } from '@aws-amplify/api';
+import Amplify, { Storage } from 'aws-amplify';
+import awsconfig from './aws-exports';
+import { speakTranslatedImageText } from './graphql/queries';
+
+/* Configure Exports */
+Amplify.configure(awsconfig);
+
+function SpeakTranslatedImage() {
+  const [ src, setSrc ] = useState("");
+  const [ img, setImg ] = useState("");
+  
+  function putS3Image(event) {
+    const file = event.target.files[0];
+    Storage.put(file.name, file)
+    .then (async (result) => {
+      setSrc(await speakTranslatedImageTextOP(result.key))
+      setImg(await Storage.get(result.key));
+    })
+    .catch(err => console.log(err));
+  }
+
+  return (
+    <div className="Text">
+      <div>
+        <h3>Upload Image</h3>
+        <input
+              type = "file" accept='image/jpeg'
+              onChange = {(event) => {
+                putS3Image(event)
+              }}
+          />
+        <br />
+        { img && <img src = {img}></img>}
+        { src && 
+          <div> <audio id="audioPlayback" controls>
+              <source id="audioSource" type="audio/mp3" src = {src}/>
+          </audio> </div>
+        }
+      </div>
+    </div>
+  );
+}
+
+async function speakTranslatedImageTextOP(key) {
+  const inputObj = { 
+    translateText: { 
+      sourceLanguage: "en", targetLanguage: "es" }, 
+    identifyText: { key },
+    convertTextToSpeech: { voiceID: "Conchita" } 
+  };
+  const response = await API.graphql(
+    graphqlOperation(speakTranslatedImageText, { input: inputObj }));
+  return response.data.speakTranslatedImageText;
+}
+function App() {
+  return (
+    <div className="App">
+        <h1>Speak Translated Image</h1>
+        < SpeakTranslatedImage />
+    </div>
+  );
+}
+export default App;
+```
+
+#### How it works
+From example schema above, `@predictions` will create resources to communicate with Amazon Rekognition, Translate and Polly.
+For each action the following is created: 
+
+- IAM Policy for each service (e.g. Amazon Rekognition `detectText` Policy)
+- An AppSync VTL function
+- An AppSync DataSource
+
+Finally a resolver is created for `speakTranslatedImageText` which is a pipeline resolver composed of AppSync functions which are defined by the action list provided in the directive.
+
+#### Actions
+Valid Actions allowed in predictions are as follows:
```

Comment:
We should also add that these 4  actions can be called individually as well. Right now we mention just convertTextToSpeech alone